### PR TITLE
feat: add command-response sdk

### PR DESCRIPTION
**Pull request by dooriya**
Add command-and-response support into Notification SDK to allow developers:
- easily implement a command handler by following the `TeamsFxBotCommandHandler` interface.
- easily register the a  `TeamsFxBotCommandHandler` instance using the `ConversationBot.initialize` API.
- easily build a combined scenario with both notification and command-response enabled.

TODO: rename the `sdk/src/notification` folder name to `sdk/src/conversation`.

### CODE REVIEW

File: packages/sdk/src/index.ts [hunk 1]
```diff
@@ -32,12 +32,11 @@ export { IdentityType } from "./models/identityType";
...
-  BotNotification,
-  BotNotificationOptions,
   Channel,
   IncomingWebhookTarget,
   Member,
   sendAdaptiveCard,
   sendMessage,
   TeamsBotInstallation,
 } from "./notification/notification";
+export { ConversationBot, ConversationOptions } from "./notification/conversation";
```
Comment (swatDong) [inline packages/sdk/src/index.ts:42]: I'm considering to rename the folder name... since it's not for `notification` only.

Comment (dooriya) [reply packages/sdk/src/index.ts:42]: This is already added to my TODO, see the PR description. I didn't rename the folder name here is for better PR review

Comment (swatDong) [reply packages/sdk/src/index.ts:42]: E.g., to `conversation/`

**Review by swatDong — APPROVED** [review]
The code logic looks good to me.

**Pull request merged**